wpengine/lostromos
# Added support to pull chart from private repo

Signed-off-by: rj.natarajan <nraddy@gmail.com>

# What Are We Doing Here

Closes #60 

Added support to pull charts from private repositories. Used helm chat downloader to download the charts to local directory. If that fails, the flow falls back to using the chart from the configured directory when the controller was created or the previous version (if it was already downloaded).

The versioned chart is stored inside folder named with "base64 encoded versioned string" under temp directory. The change presumes that the private repo is added to the lostromos pod outside of the code, possibly during bootstrap.

## How to Verify

1. Check out this PR
2. make clean all
3. Profit
